integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -231,73 +240,36 @@ object SecondaryIndexUtil {
           } else {
             siRebuildRDD.partitions.foreach { partition =>
               val carbonSparkPartition = partition.asInstanceOf[CarbonSparkPartition]
-              deleteOldCarbonDataFiles(carbonSparkPartition)
+              deleteOldCarbonDataFiles(carbonSparkPartition, validSegmentsToUse.toList)
             }
           }
+          val segmentToLoadStartTimeMap: scala.collection.mutable.Map[String, java.lang.Long] =
+            scala.collection.mutable.Map()
+          // merge index files and write segment file for merged segments
           mergedSegments.asScala.map { seg =>
+            val segmentPath = CarbonTablePath.getSegmentPath(tablePath, seg.getLoadName)
+            try {
+              new CarbonIndexFileMergeWriter(indexCarbonTable)
+                .writeMergeIndexFileBasedOnSegmentFolder(null, false, segmentPath,
+                  seg.getLoadName, carbonLoadModel.getFactTimeStamp.toString,
+                  true)
+            } catch {
+              case e: IOException =>
+                val message =
+                  "Failed to merge index files in path: " + segmentPath + ": " + e.getMessage()
```

Comment:
its a scala class, so you can use `s"..... $variable ...."` for string concatination

Reply:
Done